app/controllers/admin/facilities_controller.rb

```diff
@@ -5,44 +5,35 @@ class Admin::FacilitiesController < AdminController
 
   before_action :set_facility, only: [:show, :edit, :update, :destroy]
   before_action :set_facility_group, only: [:show, :new, :create, :edit, :update, :destroy]
+  before_action :authorize_facility, only: [:show, :new, :create, :edit, :update, :destroy]
   before_action :initialize_upload, :validate_file_type, :validate_file_size, :parse_file,
     :validate_facility_rows, if: :file_exists?, only: [:upload]
 
   def index
-    authorize([:manage, :facility, Facility])
+    authorize([:upcoming, :manage, Facility], :allowed?)
 
-    if searching?
-      facilities = policy_scope([:manage, :facility, Facility]).search_by_name(search_query)
-      facility_groups = FacilityGroup.where(facilities: facilities)
+    facilities =
+      if searching?
+        policy_scope([:upcoming, :manage, Facility]).search_by_name(search_query)
+      else
+        policy_scope([:upcoming, :manage, Facility])
+      end
 
-      @organizations = Organization.where(facility_groups: facility_groups)
-      @facility_groups = facility_groups.group_by(&:organization)
-      @facilities = facilities.group_by(&:facility_group)
-    else
-      @organizations = policy_scope([:manage, :facility, Organization])
-      @facility_groups = @organizations.map { |organization|
-        [organization, policy_scope([:manage, :facility, organization.facility_groups])]
-      }.to_h
-      @facilities = @facility_groups.values.flatten.map { |facility_group|
-        [facility_group, policy_scope([:manage, :facility, facility_group.facilities])]
-      }.to_h
-    end
+    facility_groups = FacilityGroup.where(facilities: facilities)
+    @organizations = Organization.where(facility_groups: facility_groups)
```

Comment:
It's unclear to me what `orgs = current_admin.managed_organizations` is supposed to mean.

Is it supposed to mean,

1. all the organizations you can read even if you have Facility-level write permissions?
2. all the organizations you can read if you only have Org-level write permissions?

If it's 1.,

Then I don't think the [Take 3](https://github.com/simpledotorg/simple-server/pull/1173) queries will work here because you don't walk down the tree (Org -> FG -> F).

Even if it does that, it would be very confusing because you can't really **manage** those organizations if you only have Facility level permissions.

If it's 2.,

Then I don't think this page will work because this page needs to show all the facilities the user has access to and then group them by facility groups and then organizations; regardless of what level of write permissions they are at.

Reply:
Looks like we fetch a list of orgs just to bucket the way we render a list of authorized facilities on this page. We could do the same thing in a less twitchy way with `facilities.includes(facility_group: :organization).flat_map(&:organization)`, since `facilities` is an authorized list

Reply:
In other words, we're not looking to fetch an authorized list of organizations here, we're just enriching an authorized list of facilities with org data so we can render it all fancy.

Reply:
> It's unclear to me what `orgs = current_admin.managed_organizations` is supposed to mean.
> 
> Is it supposed to mean,
> 
>     1. all the organizations you can read even if you have Facility-level write permissions?
> 
>     2. all the organizations you can read if you only have Org-level write permissions?

Hm, maybe this is part of the disconnect here. If you are asking a current admin for the organizations they manage, its literally just that: what orgs does this user have the ability to manage (i.e update and write to).  If they can manage an org, the reading is implicit, because of our auth model says that manage rights include read rights on that particular region.

The idea behind a `managed_organizations` scope is to _only_ return orgs that a user manages for the purposes of allowing them to add facility groups under that org, or to allow them to edit the org metadata, or whatever.

I may have misunderstood what this specific page is doing, and this kind of scope would probably be better if this page lets them maintain facilities they have access to:

```
current_user.managed_facilities
```

On the other hand, if the page is more for display / reading, or if we want to return _all_ facilities a user can read and then determine edit level access in the view, we could do this:

```
current_user.readable_facilities
```

or just
```
current_admin.facilities
```

Then in the view, psuedocode like this:

```
facilities.each do |facility|
  if current_admin.can_manage?(facility)
    <%= form_for facility ... %>
  else
    # just do the read-only logic
    <%= facility.name %> ...
  end
end
```

Does that help clear things up?  The point is that the outer domain layer itself can handle returning the appropriate scoped objects, and that is all that developers should have to worry about when writing new controllers or when updating actions. Nothing about `Facility.where` or `Organization.where` tells me how records should be scoped to an admin, where as an api like `current_admin.get_me_the_things_I_can_manage` or `current_admin.get_me_things_I_have_read_access_to` does.